hack/images/dcos/functions.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/sh
+
+test -n "$sandbox" || die failed to identify mesos sandbox
+
+die() {
+  test ${#} -eq 0 || echo "$@" >&2
+  exit 1
+}
+
+lookup_ip() {
+  local leader=$(nslookup "$1" | sed -e '/^Server:/,/^Address .*$/{ d }' -e '/^$/d'|grep -e '^Address '|cut -f3 -d' '|head -1)
+  test -n "$leader" || die Failed to identify $1
+  echo $leader
+}
+
+leading_master_ip() {
+  test -n "$K8SM_MESOS_MASTER" && {
+    echo $K8SM_MESOS_MASTER
+    return
+  }
+  local leader=$(nslookup leader.mesos | sed -e '/^Server:/,/^Address .*$/{ d }' -e '/^$/d'|grep -e '^Address '|cut -f3 -d' ')
+  test -n "$leader" || die Failed to identify mesos master, missing K8SM_MESOS_MASTER variable and cannot find leader.mesos
+  echo leader.mesos
+}
+
+prepare_var_run() {
```

Comment:
NOTE: this is really intended only for the scheduler since it runs in a docker and has r/w access to the root file system, unlike the executor and proxy services.